src/background/index.ts

```diff
@@ -0,0 +1,76 @@
+import '@/lib/initPolyfills';
+import { PopupActionType } from '@/types';
+import { openPopup, removePopup, getPopup } from './bgPopupHandler';
+import updateDynamicRules from './redirectRule';
```

Comment:
As fart as I see thif function is used only in this file, so maybe it would be good idea to just move it here?

Reply:
It's also used in `offscreen.ts`